diff --git a/src/Quarks.c b/src/Quarks.c
index 0dd91968..dbaf5086 100644
--- a/src/Quarks.c
+++ b/src/Quarks.c
@@ -105,6 +105,11 @@ static XrmQuark nextUniq = -1;	/* next quark from XrmUniqueQuark */
 #define CLEARPERM(q) BYTEREF(q) &= ~(1 << ((q) & 7))
 #endif
 
+#undef _XLockMutex
+#undef _XUnlockMutex
+#define _XLockMutex(m)
+#define _XUnlockMutex(m)
+
 /* Permanent memory allocation */
 
 #define WALIGN sizeof(unsigned long)
diff --git a/src/Xrm.c b/src/Xrm.c
index a8ebf883..d716fdb0 100644
--- a/src/Xrm.c
+++ b/src/Xrm.c
@@ -301,6 +301,13 @@ typedef unsigned char XrmBits;
 #define is_simple(bits)		((bits) & (NORMAL|BSLASH))
 #define is_special(bits)	((bits) & (ENDOF|BSLASH))
 
+#undef _XLockMutex
+#undef _XUnlockMutex
+#undef _XCreateMutex
+#define _XLockMutex(m)
+#define _XUnlockMutex(m)
+#define _XCreateMutex(m)
+
 /* parsing types */
 static XrmBits const xrmtypes[256] = {
     EOS,0,0,0,0,0,0,0,
@@ -500,9 +507,7 @@ static XrmDatabase NewDatabase(void)
 	_XCreateMutex(&db->linfo);
 	db->table = (NTable)NULL;
 	db->mbstate = (XPointer)NULL;
-	db->methods = _XrmInitParseInfo(&db->mbstate);
-	if (!db->methods)
-	    db->methods = &mb_methods;
+	db->methods = &mb_methods;
     }
     return db;
 }
--- a/src/KeyBind.c
+++ b/src/KeyBind.c
@@ -70,2 +70,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 static void
 ComputeMaskFromKeytrans(
@@ -288,2 +288,4 @@
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
+
 static void
 UCSConvertCase( register unsigned code,
@@ -788,2 +788,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 int
 _XTranslateKey(	register Display *dpy,
@@ -1040,1 +1040,3 @@
 }
+
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
--- a/src/xkb/XKBGeom.c
+++ b/src/xkb/XKBGeom.c
@@ -251,2 +251,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 static Status
 _XkbReadGeomProperties(XkbReadBufferPtr buf,
@@ -722,1 +722,3 @@
 }
+
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
--- a/src/Xrm.c
+++ b/src/Xrm.c
@@ -337,2 +337,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 XrmDatabase XrmGetDatabase(
     Display *display)
@@ -361,2 +361,4 @@
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
+
 void
 XrmStringToQuarkList(
@@ -759,2 +759,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 void XrmCombineDatabase(
     XrmDatabase	from, XrmDatabase *into,
@@ -815,2 +815,4 @@
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
+
 /* store a value in the database, overriding any existing entry */
 static void PutEntry(
@@ -1671,2 +1671,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 Status
 XrmCombineFileDatabase(
@@ -1697,2 +1697,4 @@
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
+
 /* call the user proc for every value in the table, arbitrary order.
  * stop if user proc returns True.  level is current depth in database.
@@ -2632,2 +2632,4 @@
+#ifndef XKBCOMP_NO_XLIB_PROTOCOL
+
 void XrmDestroyDatabase(
     XrmDatabase   db)
@@ -2656,1 +2656,3 @@
 }
+
+#endif /* XKBCOMP_NO_XLIB_PROTOCOL */
